### PR TITLE
Reorganize & implement ST queue

### PR DESCRIPTION
**Pull request by Qaysed**
Functional changes:
- add storage of ST queue
- add commands to handle ST queue: Enqueue, Dequeue, MoveDown; for mods only: InitQueue, KickFromQueue, MoveToSpot
- update HelpMe
- OffServerArchive creates a discussion thread after finishing the archival
- various minor changes made in process of refactoring
- updated Readme for changed deployment process

Technical changes:
- introduce utility file with Helper class to handle common tasks
- extract most commands into Cogs for improved clarity
- handling of file storage
- lots of general refactoring

### CODE REVIEW

File: Cogs/Queue.py [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+        await self.update_queue_message(self.queues[channel_type])
+        with open(self.QueueLocation, "w") as f:
+            json.dump(self.queues, f)
+
+        await self.helper.finish_processing(ctx)
+        print("-= The MoveDown command was used successfully by " + str(ctx.author.name) + " at " + str(
+            strftime("%a, %d %b %Y %H:%M:%S ", gmtime()) + "=-"))
+
+    @commands.command()
+    async def KickFromQueue(self, ctx: commands.Context, member: nextcord.Member):
```
Comment (JackKBroome) [inline Cogs/Queue.py]: I'd possibly consider changing the name from 'Kick' to 'Remove' as kick has moderator connotations which might not always be correct to invoke, small change but one I think might be worth changing.

File: Cogs/Queue.py [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+            await self.helper.finish_processing(ctx)
+            print("-= The InitQueue command was used successfully by " + str(ctx.author.name) + " at " + str(
+                strftime("%a, %d %b %Y %H:%M:%S ", gmtime()) + "=-"))
+        else:
+            await utility.deny_command(ctx, "InitQueue")
+            await utility.dm_user(ctx.author, "This command is restricted to moderators")
+        await self.helper.log(f"{ctx.author.mention} has run the InitQueue command in {ctx.channel.mention}")
+
+    @commands.command()
+    async def Enqueue(self, ctx: commands.Context, channel_type: ChannelTypeParameter, script: str, availability: str,
```
Comment (JackKBroome) [inline Cogs/Queue.py]: There are some small plans for a text & live queue to work from the same Bot, so I'd be more inclined to have the command be something along the "TextQueueJoin", not sold on that being the best name though, any ideas welcome

File: Cogs/Queue.py [hunk 3]
```diff
@@ -0,0 +1,291 @@
...
+            print("-= The Enqueue command was used successfully by " + str(ctx.author.name) + " at " + str(
+                strftime("%a, %d %b %Y %H:%M:%S ", gmtime()) + "=-"))
+        else:
+            await utility.deny_command(ctx, "Enqueue")
+            await utility.dm_user(ctx.author, "You may not join a text ST queue while you are already in one")
+
+        await self.helper.log(f"{ctx.author.mention} has run the Enqueue command")
+
+    @commands.command()
+    async def Dequeue(self, ctx: commands.Context):
```
Comment (JackKBroome) [inline Cogs/Queue.py]: There are some small plans for a text & live queue to work from the same Bot, so I'd be more inclined to have the command be something along the "TextQueueLeave", not sold on that being the best name though, any ideas welcome

File: Cogs/Other.py [hunk 4]
```diff
@@ -0,0 +1,289 @@
...
+                               value='Grants you the ST role of the given game, unless it is already occupied\n' +
+                                     'Usage examples: `>ClaimGrimoire 1`, `>ClaimGrimoire x3`',
+                               inline=False)
+        anyone_embed.add_field(name=">Enqueue [channel type] [script name] [availability] [notes (optional)]",
+                               value="Adds you to the queue for the given channel type (regular/experimental), "
+                                     "listing the provided information, unless you are already in either of the "
+                                     "queues. Do not join a queue if you are currently storytelling. Note that if a "
+                                     "parameter contains spaces, you have to surround it with quotes.\n" +
+                                     'Usage examples: `>Enqueue regular "Trouble Brewing" "anytime after june"`, '
+                                     '`>Enqueue Exp "Oops All Amnesiacs" "between 07-13 and 07-30" "Let me know '
```
Comment (JackKBroome) [inline Cogs/Other.py]: American dates >-<

File: Cogs/Other.py [hunk 5]
```diff
@@ -0,0 +1,289 @@
...
+                                     "no-mod`. Default is `no-mod`\n"
+                                     "Usage example: `>HelpMe all`",
+                               inline=False)
+        anyone_embed.set_footer(text="1/3")
+
+        st_embed = nextcord.Embed(title="Unofficial Text Game Bot",
+                                  description="Commands that can be executed by the ST of the relevant game - mods "
+                                              "can ignore this restriction",
+                                  color=0xe100ff)
+        st_embed.set_thumbnail(url="https://wiki.bloodontheclocktower.com/images/6/67/Thief_Icon.png")
```
Comment (JackKBroome) [inline Cogs/Other.py:163]: Link currently Broken, will find a replacement

File: Cogs/Other.py [hunk 6]
```diff
@@ -0,0 +1,289 @@
...
+                           inline=False)
+        st_embed.add_field(name=">DropGrimoire [game number]",
+                           value='Removes the ST role for the game from you\n' +
+                                 'Usage examples: `>DropGrimoire 1`, `>DropGrimoire x3`',
+                           inline=False)
+        st_embed.add_field(name=">ShareGrimoire [game number] [User]",
+                           value='Gives the ST role for the game to the given user without removing it from you. Use '
+                                 'if you want to co-ST a game.You can provide a user by ID, mention/ping, '
+                                 'or nickname, though giving the nickname may find the wrong user.\n' +
+                                 'Usage examples: `>ShareGrimoire 1 @Daddy Ben`, `>ShareGrimoire x3 108309184147185664`',
```
Comment (JackKBroome) [inline Cogs/Other.py]: Add an Underscore in usernames to show it's not 2 different inputs

File: Cogs/Other.py [hunk 7]
```diff
@@ -0,0 +1,289 @@
...
+                           value='Removes the appropriate game role from the given users. You can provide a user by '
+                                 'ID, mention/ping, or nickname, though giving the nickname may find the wrong '
+                                 'user.\n' +
+                                 'Usage examples: `>RemoveKibitz 1 793448603309441095`, `>RemoveKibitz x3 @eevee @Pam @Velvet`',
+                           inline=False)
+        st_embed.set_footer(text="2/3")
+
+        mod_embed = nextcord.Embed(title="Unofficial Text Game Bot",
+                                   description="Commands that can only be executed by moderators", color=0xe100ff)
+        mod_embed.set_thumbnail(url="https://wiki.bloodontheclocktower.com/images/6/67/Thief_Icon.png")
```
Comment (JackKBroome) [inline Cogs/Other.py:246]: Same dead link issue

File: Carat.py [hunk 8]
```diff
@@ -0,0 +1,47 @@
+import os
+
+import nextcord
+from dotenv import load_dotenv
+from nextcord.ext import commands
+
+from Cogs.Game import Game
+from Cogs.Grimoire import Grimoire
+from Cogs.Other import Other
+from Cogs.Queue import Queue
```
Comment (JackKBroome) [inline Carat.py]: Can the cog name be changed to TextQueue as Live queue is being considered

File: Cogs/Other.py (hunk 4 above)
Comment (Qaysed) [reply Cogs/Other.py]: yeah 😅 gonna change to something more locale agnostic

**Review by JackKBroome — APPROVED** [review]

**Pull request merged**